### ISSUE TITLE
.d.ts Errors

### ISSUE THREAD

**Issue by kadiryazici** [johnfn/ts2gd#55]
I get .d.ts errors when using `ts2gd`. For example I created a file src/Player.ts and wrote something to it then saved, `ts2gd` throws an error `Player.d.ts` not found and stops compiling.

**Comment by johnfn** [comment]
Wow, that's not good! A couple of questions:

* Does this happen on a new project with just Player.ts?
* Could you paste the whole output (or a screenshot)? 
* What OS are you running?
* Ideally, could you create a github repo that reproduces this failure so I can pull it down and investigate?

**Comment by kadiryazici** [comment]
@johnfn Okay I'm creating it.

**Comment by kadiryazici** [comment]
`OS:` Windows 11,
`Godot Version:` Download from steam today.
`Graphics Card:` Ryzen 5 4600h Graphics Processor + Gtx 1650 Ti
`Processor:` Ryzen 5 4600h
`Node Version:` 16.13.0
`Repo:` https://github.com/kadiryazici/ts2gd.d.ts.error

**Comment by johnfn** [comment]
Thanks! And finally, mind pasting the exact output error that you see in the terminal?

**Comment by johnfn** [comment]
Ahh, OK, this appears to be a Windows issue. Problem is I develop mostly on Mac so I don't have a ton of time to test on Windows. I'll try to look into this soon however :)

**Comment by ksjogo** [comment]
I think Github Actions have Windows as virtual environment now. We could potentially set up some automated tests.

**Comment by kadiryazici** [comment]
I wish I could test and see what's wrong but the code is beyond my skills :D I understand nothing of the code.

**Comment by ksjogo** [comment]
Popping that into the launch.json:
```
{
      "name": "Launch Generate Defs",
      "program": "${workspaceFolder}/js/main.js",
      "request": "launch",
      "skipFiles": [
        "<node_internals>/**"
      ],
      "args": [
        "--buildLibraries",
        " --buildOnly",
        "${workspaceFolder}/dummy/ts2gd.json"
      ],
      "type": "pwa-node",
      "preLaunchTask": "tsc: build - tsconfig.json",
      "outFiles": [
        "${workspaceFolder}/js/**/*.js"
      ]
    }
    ```
    And then running that target should get you a debugger stopping at the crash.

**Comment by ksjogo** [comment]
I started reworking path managment in #56 
I can do some testing on Windows between the years.

**Comment by johnfn** [comment]
The repo he provided has the stack trace: https://github.com/kadiryazici/ts2gd.d.ts.error

Problem is I think there will be many windows issues, not just this one, so I really just need to test it on Windows. I'll get a VM :)

**Comment by TTSKarlsson** [comment]
It's about the paths as you assumed. Specifically the backslashes, but I would think the colon also is a problem.

![image](https://user-images.githubusercontent.com/9962943/147484977-e8958197-25c5-4320-adf7-b3e8f8b3fa92.png)

That's the @asset_paths.d.ts that was generated/copied. On windows the backslashes needs to at least be escaped in within the quotes.

**Comment by ksjogo** [comment]
I am looking into adding Windows to the CI in https://github.com/johnfn/ts2gd/pull/81
We need some integration test for full generation/typecheck still.
Should be doable after adding jest.

**Comment by ksjogo** [comment]
Got it to compile run my game on Windows now with the code on #81.
Currently has some failing tests, but they will be okay when I rebase on project-based tests.

Had to rework most of the path handling in the asset classes though.
And the PR is needing some of my other MRs first.

tldr; Windows will work soon.
